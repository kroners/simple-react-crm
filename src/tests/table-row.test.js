import { fireEvent, render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import TableRow from '../components/table-row';

it('Render TableRow component and match snapshot', () => {
  const data = { id: 'abc', name: 'Demo' };
  const component = renderer.create(<TableRow data={data} />);
  let tree = component.toJSON();
  console.log(tree.props);
  expect(tree).toMatchSnapshot();
});

it('Renders rows from data and onValidate function', () => {
  const data = { id: 'abc', name: 'Demo' };
  const onValidateMock = jest.fn();

  render(<TableRow data={data} onValidate={onValidateMock} />);

  expect(screen.getByText('abc')).toBeInTheDocument();
  expect(screen.getByText('Demo')).toBeInTheDocument();

  const button = screen.getByRole('button');
  fireEvent.click(button);

  expect(onValidateMock).toHaveBeenCalled();
});
