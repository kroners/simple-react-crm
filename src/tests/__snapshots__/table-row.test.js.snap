// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render TableRow component and match snapshot 1`] = `
<tr>
  <td>
    abc
  </td>
  <td>
    Demo
  </td>
  <td>
    <button
      onClick={[Function]}
    >
      Validate
    </button>
  </td>
</tr>
`;
